api/users.go

```diff
@@ -0,0 +1,138 @@
+			log.Event(ctx, content.ValidEmailErrorField, log.ERROR)
+			errorList = append(errorList, apierrors.IndividualErrorBuilder(apierrors.ErrInvalidEmail, apierrors.InvalidErrorMessage, content.ValidEmailErrorField, content.ValidEmailErrorParam))
+		}
+
+		// report validation errors in response
+		if len(errorList) != 0 {
+			apierrors.WriteErrorResponse(ctx, w, http.StatusBadRequest, apierrors.ErrorResponseBodyBuilder(errorList))
+			return
+		}
+
+		newUser, err := CreateNewUserModel(ctx, username, tempPassword, email, api.UserPoolId)
+		if err != nil {
+			log.Event(ctx, content.NewUserModelErrorMessage, log.ERROR)
+			apierrors.HandleUnexpectedError(ctx, w, err, content.NewUserModelErrorMessage, content.NewUserModelErrorField, content.NewUserModelErrorParam)
+			return
+		}
+
+		//Create user in cognito - and handle any errors returned
+		resultUser, err := api.CognitoClient.AdminCreateUser(newUser)
+		if err != nil {
+			if strings.Contains(err.Error(), content.InternalErrorException) {
+				log.Event(ctx, content.AdminCreateUserErrorMessage, log.ERROR)
+				apierrors.HandleUnexpectedError(ctx, w, err, content.AdminCreateUserErrorMessage, content.AdminCreateUserErrorField, content.AdminCreateUserErrorParam)
+				return
+			} else {
+				log.Event(ctx, err.Error(), log.ERROR)
+				errorList = append(errorList, apierrors.IndividualErrorBuilder(err, err.Error(), content.AdminCreateUserErrorField, content.AdminCreateUserErrorParam))
+				apierrors.WriteErrorResponse(ctx, w, http.StatusBadRequest, apierrors.ErrorResponseBodyBuilder(errorList))
+				return
+			}
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		jsonResponse, err := json.Marshal(resultUser)
+		if err != nil {
+			log.Event(ctx, content.MarshallingNewUserErrorMessage, log.ERROR)
+			apierrors.HandleUnexpectedError(ctx, w, err, content.MarshallingNewUserErrorMessage, content.MarshallingNewUserErrorField, content.MarshallingNewUserErrorParam)
+			return
+		}
+
+		w.WriteHeader(http.StatusCreated)
+		_, err = w.Write(jsonResponse)
+		if err != nil {
+			log.Event(ctx, content.HttpResponseErrorMessage, log.ERROR)
+			apierrors.HandleUnexpectedError(ctx, w, err, content.HttpResponseErrorMessage, content.HttpResponseErrorField, content.HttpResponseErrorParam)
+			return
+		}
+	}
+}
+
+//CreateNewUserModel creates and returns AdminCreateUserInput
+func CreateNewUserModel(ctx context.Context, username string, tempPass string, emailId string, userPoolId string) (*cognitoidentityprovider.AdminCreateUserInput, error) {
+	// Return an error if empty id was passed.
+	if userPoolId == "" {
+		return nil, errors.New(content.UserPoolIdNotFoundMessage)
+	}
+
+	var (
+		deliveryMethod, emailAttrName string = "EMAIL", "email"
+	)
+
+	user := &models.CreateUserInput{
+		UserInput: &cognitoidentityprovider.AdminCreateUserInput{
+			UserAttributes: []*cognitoidentityprovider.AttributeType{
+				{
+					Name:  &emailAttrName,
+					Value: &emailId,
+				},
+			},
```

Comment:
This is only setting the email attribute for the user, not their forename and surname.